emmet-core/emmet/core/electrode.py

```diff
@@ -0,0 +1,136 @@
+""" Core definition of an Electrode Document """
+from typing import List, Dict, ClassVar
+from datetime import datetime
+
+from pydantic import BaseModel, Field
+
+from emmet.stubs import ComputedEntry
+from emmet.core.structure import StructureMetadata
+
+
+class ElementEvolution(BaseModel):
+    """
+    Result of the PhaseDiagram.get element_profile which moves a along a line in
+    composition space and return the reaction, chemical potential, and the remaining
+    amount of the element
+    """
+
+    reaction: str = Field(
+        ...,
+        description="Evolution reaction determined by a facet of the phase diagram",
+    )
+
+    chempot: float = Field(
+        ...,
+        description="The chemical potential of the target element solved on that facet",
+    )
+
+    evolution: float = Field(
+        ..., description="Remaining amount of element .. TODO get better description",
+    )
+
+
+class VoltagePair(BaseModel):
+    """
+    Based on AbstractVoltagePair from pymatgen
```

Comment:
Do you want this to be a stub so you can just use `AbstractVoltagePair` in the document as both pymatgen object and a pedantic model or do you want a `VoltagePair` document type? If the latter, is this how you want to structure the data?

Reply:
OK, I can do the former.
I'm not so sure I understand how the stubs work though